### PR TITLE
Bug 1527853 - Refactor update_translation view

### PR DESCRIPTION
**Pull request by adngdb**
This moves the `update_translation` view into a simpler `create_translation` view, effectively separating the actions of creating a translation and approving a translation. Thus I was able to remove some logic that was not needed anymore.

This also moves the `delete_translation` to the `translations` app, as well as a bunch of tests.

Note that we have a `translate` app and a `translations` app. The former is for serving the Translate page, the latter has logic for handling the `Translation` model (create and delete). I originally didn't intend to do such a separation, but it was needed in order for URLs to work, and in the end I think it makes a lot more sense. In the future, we might want to move the actual `Translation` model into this app.

### CODE REVIEW

**Comment by mathjazz** [comment]
I finished a high-level view of the code and it looks good! Thanks!

I have yet to test the code locally, but before I do so... Shouldn't we merge the translations and the `review` app, which is also "handling the `Translation`" model?

**Comment by adngdb** [comment]
Mmm, that's a good question. I like having the review actions in a separate app, but maybe that doesn't make sense? It really depends on how we decide to effectively organize those apps. Right now, there's no real logic, as far as I can tell. 

We can split based on models, one app per model or group of models, in which case the review actions should go with the translations app. Or we can split by functionality, one app per page or feature, in which case it's better to keep them separate.

I'm not too sure, but I know that I would rather have separate modules, so I'd vote for keeping them as they are for the moment, and revisit when the structure of the apps makes more sense.

**Comment by mathjazz** [comment]
Yeah, we need to rethink how we split code into modules, because we broke the original idea long time ago. I worry that having separate `translations` and `translate` views just adds to confusion. 

Also, if you're familiar with the `review` module (which contains review actions), you'd expect translation creation actions in a view called `translate`, not `translation`.

I'm fine with leaving this for a later refactor.

**Comment by adngdb** [comment]
Rebased and ready! :birthday:

File: pontoon/translations/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+from pontoon.checks.models import Warning
+from pontoon.test.factories import TranslationFactory
+
+
+@pytest.yield_fixture
+def request_create_translation():
+    """
+    Return a function to call the create_translation view with default parameters.
+    """
+
```
[the remark this answers is not in the document]
Comment (adngdb) [reply pontoon/translations/tests/test_views.py]: Nope, that's a black thing. :innocent:

**Comment by adngdb** [comment]
@mathjazz I've merged `review` into `translations`. I do like that `translate` and `translations` are separated, as they effectively do two very different things. I've also added README files to explain what each app does.

**Pull request merged**